sdks/router-sdk/src/utils/isValidTokenPath.ts

```diff
@@ -0,0 +1,25 @@
+import { Currency, Token } from '@uniswap/sdk-core'
+import { Pool as V4Pool } from '@uniswap/v4-sdk'
+import { Pair } from '@uniswap/v2-sdk'
+import { Pool as V3Pool } from '@uniswap/v3-sdk'
+
+type TPool = Pair | V3Pool | V4Pool
+
+export function isValidTokenPath(prevPool: TPool, currentPool: TPool, inputToken: Currency): boolean {
+  if (currentPool.involvesToken(inputToken as Token)) return true
```

Comment:
this should always be true if inputToken is not native right, a comment could be helpful

Reply:
I have to put `as Token` bc v2/v3 pools only accept token parameters..so I'm basically hacking typescript so it will compile. this will work if the inputToken IS native too....